src/static/static.controller.ts

```diff
@@ -0,0 +1,44 @@
+import { Controller, Get, Logger, Param, Res } from '@nestjs/common';
+import { Response } from 'express';
+import { ApiOkResponse, ApiTags } from '@nestjs/swagger';
+import { isHddStaticServiceConfigured, isS3ServiceConfigured } from './utils';
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+
+@ApiTags('images')
+@Controller('images')
+export class StaticController {
+  private readonly logger: Logger = new Logger(StaticController.name);
+
+  @Get('/:fileName')
+  @ApiOkResponse()
+  async getUrlAndRedirect(@Param('fileName') fileName: string, @Res() res: Response) {
+    try {
+      if (isHddStaticServiceConfigured()) {
+        res.redirect('/' + fileName);
+      }
+      if (isS3ServiceConfigured()) {
+        const AWS_ACCESS_KEY_ID = process.env.AWS_ACCESS_KEY_ID;
+        const AWS_SECRET_ACCESS_KEY = process.env.AWS_SECRET_ACCESS_KEY;
+        const AWS_REGION = process.env.AWS_REGION;
+        const AWS_S3_BUCKET_NAME = process.env.AWS_S3_BUCKET_NAME;
+
+        const s3Client = new S3Client({
```

Comment:
let's move this logic to service layer like `getImageUrl` for both hdd and s3

Reply:
Thanks. Done.